distributed/worker.py

```diff
@@ -1718,7 +1844,19 @@ def transition_ready_waiting(self, ts):
         """
         This transition is common for work stealing
         """
-        pass
+        ts.runspec = None
+
+    def transition_waiting_new(self, ts):
```

Comment:
is a transition back to `new` actually allowed? I consider `new` as a pseudo/virtual state but I wouldn't expect a transition to actually move a task to `new`, with the sole exception of a "catch all" mechanism, similar to the scheduler task state machine where the `start -> release -> * -> finish` is a a catch all transition path but we haven't implemented this kind of thing for the worker, yet.

Reply:
Yeah, this is a leftover I forgot to remove, it's not reachable via the `_transitions` dict.  I was starting down the state machine path and pulled back to leave that for a later PR.  I'll rip this out.  Thanks!